From a62dc59a236c945aad353c613c01dd5f5b5ad801 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@videolabs.io>
Date: Tue, 17 Jan 2017 14:17:20 +0100
Subject: [PATCH 12/39] direct3d11: the D3D11 debug should not depend on MSVC
 building

Just on the presence of the SDK on the running machine.
---
 modules/video_output/win32/direct3d11.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/modules/video_output/win32/direct3d11.c b/modules/video_output/win32/direct3d11.c
index 1834828..d5b926b 100644
--- a/modules/video_output/win32/direct3d11.c
+++ b/modules/video_output/win32/direct3d11.c
@@ -1114,8 +1114,12 @@ static int Direct3D11Open(vout_display_t *vd, video_format_t *fmt)
     UINT creationFlags = 0;
     HRESULT hr = S_OK;
 
-# if !defined(NDEBUG) && defined(_MSC_VER)
-    creationFlags |= D3D11_CREATE_DEVICE_DEBUG;
+# if !defined(NDEBUG)
+    HINSTANCE sdklayer_dll = LoadLibrary(TEXT("d3d11_1sdklayers.dll"));
+    if (sdklayer_dll) {
+        creationFlags |= D3D11_CREATE_DEVICE_DEBUG;
+        FreeLibrary(sdklayer_dll);
+    }
 # endif
 
     DXGI_SWAP_CHAIN_DESC1 scd;
-- 
2.10.1.windows.1

